integrations/transportation/trainline.js

```diff
@@ -0,0 +1,149 @@
+import { get } from 'lodash';
+
+import {
+  ACTIVITY_TYPE_TRANSPORTATION,
+  TRANSPORTATION_MODE_TRAIN,
+  TRANSPORTATION_MODE_BUS,
+  TRANSPORTATION_MODE_PUBLIC_TRANSPORT,
+} from '../../definitions';
+import { ValidationError, AuthenticationError } from '../utils/errors';
+import parseCookies from '../authentication/parseCookies';
+import request from '../utils/request';
+
+const LOGIN_PATH = 'https://www.thetrainline.com/login-service/api/login';
+const PAST_BOOKINGS_PATH = 'https://www.thetrainline.com/my-account/api/bookings/past';
+
+function matchTransportMode(modeFromTrainline) {
+  switch (modeFromTrainline) {
+    case 'train':
+      return TRANSPORTATION_MODE_TRAIN;
+    case 'bus':
+      return TRANSPORTATION_MODE_BUS;
+    default:
+      return TRANSPORTATION_MODE_PUBLIC_TRANSPORT;
```

Comment:
maybe worth showing a warning in the logs here as this shouldn't happen right?

Reply:
Yep, I've added a `logger.logWarning()` to show that the transport mode hasn't matched